src/CloudinaryAdapter.php

```diff
@@ -34,13 +34,23 @@ public function __construct(ApiFacade $api)
      */
     public function write($path, $contents, Config $config)
     {
+        $path = $this->removeExtension($path);
+
         try {
             return $this->normalizeMetadata($this->api->upload($path, $contents));
         } catch (\Exception $e) {
             return false;
         }
     }
 
+    private function removeExtension($path)
+    {
+        $ext = pathinfo($path, PATHINFO_EXTENSION);
```

Comment:
Would be awesome to not use that substr hell. Didnt figure out how to make it cleaner, though.